src/main/scala/org/broadinstitute/hail/sparkextras/OrderedPartitioner.scala

```diff
@@ -13,22 +13,23 @@ import scala.reflect.{ClassTag, classTag}
 import scala.util.Random
 import scala.util.hashing.{MurmurHash3, byteswap32}
 
-case class OrderedPartitioner[PK, K](rangeBounds: Array[PK], ascending: Boolean = true)(implicit val kOk: OrderedKey[PK, K])
+case class OrderedPartitioner[PK, K](rangeBounds: Array[PK], ascending: Boolean = true, empty: Boolean = false)
+  (implicit val kOk: OrderedKey[PK, K])
   extends Partitioner {
 
   import kOk.pkct
   import kOk.pkOrd
   import Ordering.Implicits._
 
-  require(rangeBounds.isEmpty ||
-    rangeBounds.zip(rangeBounds.tail).forall { case (left, right) => left < right })
+  require(!empty || rangeBounds.isEmpty)
```

Comment:
Agreed, but I think by default that's only available in the test framework.  Or did we add it somewhere?

```
scala> true ==> false
<console>:8: error: value ==> is not a member of Boolean
              true ==> false
                   ^
```